index.js

```diff
@@ -926,6 +927,109 @@ PersonaClient.prototype.getProfileByGuid = function(opts, callback){
+        // if we've collected personaQueryLimit guids in the current guidList array, push it onto the guidListArray and
+        // reset for next lot
+        if (guidList.length === personaQueryLimit) {
+            guidListArray.push(guidList);
+            guidList = [];
+        }
+    });
+    // see if we have any guidList left that have not been pushed to the guidListArray yet, push them now if that's the case
+    if (guidList.length > 0) {
+        guidListArray.push(guidList);
+    }
+    var parallelFNs = [];
+    // for each of the "blocks" of users we want to hydrate, create a parallel function and add it to
+    // an array. Each of these functions will call Persona with a block of the users to hydrate...
+    guidListArray.forEach(function (guidList) {
+        var ids = '';
+        // generate a comma-sep list of user's from the list of users we need to process
+        guidList.forEach(function (guid) {
+            ids += (ids !== '') ? ',' + guid : guid;
+        });
+        parallelFNs.push(function pushFn(cb) {
+            var options = {
+                hostname: _this.config.persona_host,
+                port: _this.config.persona_port,
+                path: '/users?guids=' + ids,
+                method: "GET",
+                headers: {
+                    'Authorization': 'Bearer ' + token,
+                    'User-Agent': _this.userAgent,
+                    'X-Request-Id': xRequestId
+                }
+            };
+            var personaReq = _this.http.request(options, function personaReq(personaResp) {
+                var userString = '';
+                personaResp.on('data', function onData(chunk) {
+                    userString += chunk;
+                });
+                personaResp.on('end', function onEnd() {
+                    if (personaResp.statusCode === 200) {
+                        var data = JSON.parse(userString);
+                        var results = [];
+                        if (!_.isEmpty(data)) {
+                            if (_.isArray(data)) {
+                                results = data;
+                            } else {
+                                results.push(data);
+                            }
+                        }
+                        cb(null, results);
+                    } else {
+                        var error = new Error();
+                        error.http_code = personaResp.statusCode || 404;
+                        cb(error, null);
+                    }
+                });
+            });
+            personaReq.on('error', function onError(err) {
+                cb(err, null);
+            });
+            personaReq.on('clientError', function onClientError(err) {
+                cb(err, null);
+            });
+            personaReq.end();
+        });
+    });
+    // execute the functions in parallel - so all calls will be made to persona in parallel
+    async.parallel(parallelFNs, function (err, userArrays) {
```

Comment:
If someone calls this new client function and asks for 2000 GUIDs, this function is going to add 80 functions to execute in the event loop all at once. This feels like potential DOS abuse here. If we're really going to go down this route (see above as to me thinking not) then use `parallelLimit`